guides/data_science/deep_learning/CNN.md

```diff
@@ -0,0 +1,89 @@
+
+
+## How CNN Works?
+Input Image -> CNN -> Classified Image
+
+CNN has following layers:
+
+* Convolutional
+* ReLU Layer
+* Pooling
+* Fully Connected
+
+CNN compares the images piece by piece. The pieces that it looks for are called features.
+
+Feature Detector � usually 3*3 matrix 
+
+**Input image * feature detector = Feature Map**
+
+Feature map is the smaller version of the input image, because it is easy and faster to process the smaller image thus we perform a dot product between our image and filters.
+
+We can take more than one feature detector.
+
+
+### Convolution Layer
+
+Steps:
+1. Line up the feature and the image
+1. Multiply each image by corresponding feature pixel
+1. Add them up
+
+And now, put the value in the position of that pixel. Similarly move the feature to every position of the image. We will get a matrix, of random decimal numbers. 
+
+<img src= "http://cambridgespark.com/content/tutorials/convolutional-neural-networks-with-keras/figures/convolve.png" height = "200" width ="500"/>
+
+### ReLU Layer 
+
+In this layer, we remove the negative values from the filtered images and replaces them with 0.
+
+It is an activation function. It has value 0 until a threshold value doesn�t occur, and after that certain value it shows linear relationship.
+This is done to avoid the output 0 during the summation of matrix.
+
+f(x) = 0 if x<0 or x if x>=0
+
+<img src="https://cdn-images-1.medium.com/max/1600/1*6HyqifN4M_bJ7DTJ0RFRJA.jpeg" height="200" width="550"/>
+
+### Pooling Layer
+The use of this layer is to shrink the image into smaller size.
+
+Steps:
+1. Pick a window size (usually 2 or 3)
+1. Pick a stride � (stride is the distance by which your window moves to another pixel).
+1. Walk your window across your filtered images.
+1. From each window, take the maximum value.
+
+<img src="https://qph.fs.quoracdn.net/main-qimg-8afedfb2f82f279781bfefa269bc6a90" height="200" width="500"/>
+
+### Fully connected Layer
+This is the final layer where the actual classification happens.
+
+We take our filtered and shrinked images and put them into a single list (vector).
+
+There will be some elements which have high values for an IMAGE A and rest of the pixels will be show low values.
+
+For IMAGE B there will be some other pixels which shows high value, and rest will be close to 0.
+
+Thus, using this last vector we can classify whether the new image belong to category A or B by comparing the elements which are high in that vector.
+
+<img src="https://camo.githubusercontent.com/bb218769914fa13c8989675fef0a43eb8c1b1635/687474703a2f2f62726f687265722e6769746875622e696f2f696d616765732f636e6e31332e706e67" height="300" width="400"/> The image is classified as **X** since it has more probability. 
+
+**Combining All Layers**
+<img src="https://adeshpande3.github.io/assets/Cover.png" />
```

Comment:
Include your image sources at the end of each source properly and we are good to merge!

Reply:
I didn't get. Can u elaborate more?😅😅

Reply:
I have to mention at the bottom of the image from where I have taken this image. And I've to do this for all images..?  Is it?

Reply:
I've updated this change. Please see

Reply:
Every image you have been taking: for example if you took an image from analyticsvidhya, you need to cite it at the end as a source.

Reply:
This has been done